### ISSUE TITLE
We should not open empty editors on the variable explorer

### ISSUE THREAD

**Issue by goanpeca** [spyder-ide/spyder#9942]
We should probably not display editors on the variable explorer if the object is empty.

<img width="1120" alt="Screen Shot 2019-08-01 at 10 56 33" src="https://user-images.githubusercontent.com/3627835/62308584-336f7480-b44b-11e9-971f-acad14df1790.png">

**Comment by ccordoba12** [comment]
I think someone required that a long time ago and we implemented it (this should also be the case for empty Numpy arrays and dataframes).

So perhaps we should leave it?

**Comment by goanpeca** [comment]
>I think someone required that a long time ago and we implemented it (this should also be the case for empty Numpy arrays and dataframes).

Can't recall. The point is that unless we provide a way to add elements (which could get complicated quickly) then having an empty editor serves no purpose.

<img width="735" alt="Screen Shot 2019-08-01 at 11 16 45" src="https://user-images.githubusercontent.com/3627835/62309849-dfb25a80-b44d-11e9-8207-23cad97fe5e9.png">

**Comment by ccordoba12** [comment]
The problem is that not opening an editor for empty arrays/dataframes/collections can give users the feeling that things are not working, or that they are partially working (i.e. work for non-empty arrays and not work for empty ones).

I think that was the main reason why we added this per a user request.

**Comment by goanpeca** [comment]
Then I think we should add the ability to add things to these editors (add, insert new column, new rows). Also the editors open on selection (and not double click) which seems odd.

Opinions on adding these capabilities?

Feel free to close this one then

**Comment by ccordoba12** [comment]
> Then I think we should add the ability to add things to these editors (add, insert new column, new rows)

Hard to do it right for all kinds of arrays/collections/dataframes. I think it's better if we don't enter that path.

> Also the editors open on selection (and not double click) which seems odd.

Not for me on Linux. @dalthviz, what happens on Windows?

**Comment by jitseniesen** [comment]
Not a priority, but I think it is helpful especially for newer users if the editor indicates more clearly that it displaying an empty list / array.

**Comment by dalthviz** [comment]
> Not for me on Linux. @dalthviz, what happens on Windows?

@ccordoba12  Yep now the edition of an element triggers with single selection. I think that's because I changed the flags for the edition trigger when implementing the Variable Explorer filter [here](https://github.com/spyder-ide/spyder/blob/f63309f66fbb602539864e82247a98a8ed6df0d8/spyder/plugins/variableexplorer/widgets/collectionseditor.py#L1120). Also, there are two different behaviors when trying to trigger the edition depending on the column you click (if is the `name` one you need to double click but if it's over the value column it triggers with a single click)

**Comment by ccordoba12** [comment]
> Also, there are two different behaviors when trying to trigger the edition depending on the column you click (if is the name one you need to double click but if it's over the value column it triggers with a single click)

That's really odd. @dalthviz, please open an issue about this and unify the behavior to use double-click in all columns.

Also, please solve this ASAP so we can include it in beta4.
